msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -251,6 +251,10 @@ public void signIn(@NonNull final Activity activity,
 
             @Override
             public void onTaskCompleted(@NonNull final ILocalAuthenticationResult localAuthenticationResult) {
+                if (authenticationCallback == null) {
```

Comment:
we already check for callback `non null` before invoking the request itself right ? How are we running into this

Reply:
we check for non-null and if it is NULL, then we post the error using the `onError` of the callback itself, but since the callback itself is null, hence we can't actually post that error and run into a `NullPointerException`. I've added this check to instead throw an `IllegalStateException` which is what happens in the case of Multiple Account as well.